db/migrate/20190516141555_events_without_match.rb

```diff
@@ -0,0 +1,18 @@
+class EventsWithoutMatch < ActiveRecord::Migration
+  def up
+    add_column :match_events, :client_id, :integer
+
+    # link up existing events to the sourcing client
+    MatchEvents::Base.all.each do |event|
```

Comment:
Not sure how many `MatchEvents` are in production but this can be made *much* faster ` MatchEvents::Base.preload(:match).find_each` to automatically bulk load reasonable numbers of records (avoids bringing `all` into memory in one code) and for each patch to `preload` the match association (Preload works by running a second query "from match where event_id IN (1,2,3,4)" and then wiring up the rails models. This is in practice about 10x faster then and uses less ram then `all.each` without the preload

Reply:
No need to change of `MatchEvents.count` < 1000 or so in production

Reply:
It's probably more then 1000, but since it only runs once.... it makes sense to change it to find_each, but it doesn't need more optimization than that.

Reply:
apparently if I move from all.each to find_each, I am no longer able to use the model on line 9 (event.match). Obviously I can build the model, or work with the raw columns instead.   Is it really worth rewriting this for a onetime migration?

Reply:
Hmm, looking at the previous code, I don't see where match was being assigned.  Looks like line 9 should be `event.update(client_id: event.match.client_id)`